dashboard/modules/job/common.py

```diff
@@ -73,14 +84,15 @@ class JobInfo:
     metadata: Optional[Dict[str, str]] = None
     #: The runtime environment for the job.
     runtime_env: Optional[Dict[str, Any]] = None
+    #: The driver related to this job. For submission jobs,
+    #  it is the first driver launched by that job submission.
+    driver: Optional[DriverInfo] = None
```

Comment:
Actually, I probably shouldn't include this in this dataclass since driver is only filled out on the read path and is not ever stored in the JobInfo KV store.

Reply:
Ah good point, I guess the problem with this is that in the future someone might read a JobInfo from the internal_kv and see `driver: None` and incorrectly conclude that there's no driver?  On the other hand, maybe this is an implementation detail, and we should add it to the dataclass now in case in the future we decide to populate the `driver` field earlier.

Reply:
Yeah, this is a code smell to me that I'm unnecessarily re-using the `JobInfo` class. 

I think it's probably safer to not add it here for now and that also would make it less confusing for developers who would expect it to be filled out but it never is.